API_VERSION

```diff
@@ -0,0 +1 @@
+3.0.0
```

Comment:
Should this be in a header file instead of generating a header from this text file?
Or have the genrule that generates the header file parse the contents of this text file and do something like:

struct EnvoyVersion {
  int major;
  int minor;
  int patch;
};

EnvoyVersion envoy_version = {3, 0, 0};

Reply:
The contents of this file will be updated by tools (python code, a-la proto_format).
Although the tool can read and manipulate a C++ header file, I think it makes more sense to put the data in a separate place that both the C++ and python code can access.

Reply:
How about we write a small python script called from a genrule that generates the c++ header above from the contents of this file.

Reply:
Yeah, that could simplify code a little bit if we generate the C++ semver via a genrule script. That's true only though if we don't need to parse out string semvers for other reasons in Envoy. @adisuissa do you think we nee to go from "X.Y.Z" to struct in Envoy other than for this purpose?

Reply:
Another option is to eventually keep it in the API manifest file, and will be defined by a proto schema.
This file will also hold more data, so it might be more expensive to read, but if the read will only happen once during initialization, I think it should be ok.
I'll implement it, and see if it simplifies things.

Reply:
Are you saying we would load the manifest file at initialization?

There probably is a documentation advantage of having a plain text statement of what the current API version is, so I think the current approach actually is pretty good.

Reply:
I see. I don't think there's a need to parse the API version as a string in other places, so as @antoniovicente suggested, I'll implement a genrule to create the header file.

Reply:
SG

Reply:
@antoniovicente Thanks for the suggestion, I think it simplifies the code.
@htuch should be ready for another pass.